api/app/GraphQL/Validators/UpdatePublishedPoolInputValidator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\GraphQL\Validators;
+
+use App\Enums\PoolStatus;
+use App\Models\Pool;
+use Database\Helpers\ApiErrorEnums;
+use Illuminate\Validation\Rule;
+use Nuwave\Lighthouse\Validation\Validator;
+
+final class UpdatePublishedPoolInputValidator extends Validator
+{
+    /**
+     * Return the validation rules.
+     *
+     * @return array<string, array<mixed>>
+     */
+    public function rules(): array
+    {
+
+        $pool = Pool::find($this->arg('id'));
+
+        return [
+            'changeJustification' => [
+                Rule::when($pool->status === PoolStatus::PUBLISHED->name, [
+                    'required',
+                ]),
+            ],
+            // Optional Fields
+            'specialNote.en' => ['required_with:specialNote.fr', 'string'],
+            'specialNote.fr' => ['required_with:specialNote.en', 'string'],
+            'aboutUs.en' => ['required_with:aboutUs.fr', 'string', 'nullable'],
+            'aboutUs.fr' => ['required_with:aboutUs.en', 'string', 'nullable'],
+            'whatToExpectAdmission.en' => ['required_with:whatToExpectAdmission.fr', 'string', 'nullable'],
+            'whatToExpectAdmission.fr' => ['required_with:whatToExpectAdmission.en', 'string', 'nullable'],
+            'whatToExpect.en' => ['required_with:whatToExpect.fr', 'string', 'nullable'],
+            'whatToExpect.fr' => ['required_with:whatToExpect.en', 'string', 'nullable'],
+
+            // Always required fields
```

Comment:
```suggestion
            // Required fields
```

Reply:
This was actually intentional because they must be present where as the others are only sometimes required. The others are `nullable` as long as they both are :sweat_smile:

Reply:
Where are the "Sometimes required fields"?

Reply:
I can change the other comment to say "Sometimes required". Or just remove the comments since the rules and pretty explicit by themselves. I just copy and pasted these and left them assuming they were helping someone since they existed in the first place :stuck_out_tongue:

Reply:
:axe: [update validation for published pool mutation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10348/commits/44405540d3417d1bee8f8122c35488da3c7e3758)